lib/web/ui/user.go

```diff
@@ -76,7 +80,11 @@ func NewUserListEntry(teleUser types.User) (*UserListEntry, error) {
 
 	authType := "local"
 	if teleUser.GetUserType() == types.UserTypeSSO {
-		authType = teleUser.GetCreatedBy().Connector.Type
+		// Gracefully handle a malformed SSO user that doesn't have a "CreatedBy"
+		authType = unknownSSOAuthType
+		if connector := teleUser.GetCreatedBy().Connector; connector != nil {
+			authType = connector.Type
```

Comment:
How the `GetUserType()` can be `types.UserTypeSSO` and connector == nil  ?

The  teleUser.GetUserType() return `types.UserTypeSSO` only if ` connector != nil`


https://github.com/gravitational/teleport/blob/master/api/types/user.go#L524

```
func (u UserV2) GetUserType() UserType {
	if u.GetCreatedBy().Connector != nil ||
		len(u.GetOIDCIdentities()) > 0 ||
		len(u.GetGithubIdentities()) > 0 ||
		len(u.GetSAMLIdentities()) > 0 {

		return UserTypeSSO
	}

	return UserTypeLocal
}
```